### DISCUSSION TITLE
Improve UI around the effect of the `--now` flag

### DISCUSSION THREAD

**Discussion by ulken** [jotaen/klog#353] [Something else]
Maybe I'm missing something, but I find it surprising that **End-time** shifts as time goes by, even after the open-ended range is closed.

**Example output**

> 15:09

```sh
             Total    Should     Diff   End-Time
Today        6h35m    6h25m!     +10m      14:59 # <-- this
Other       31h10m    32h5m!     -55m
          ======== ========= ========
All         37h45m   38h30m!     -45m      15:54 # <-- and this
```

> 15:11

```sh
             Total    Should     Diff   End-Time
Today        6h35m    6h25m!     +10m      15:01 # <-- ticked two minutes
Other       31h10m    32h5m!     -55m
          ======== ========= ========
All         37h45m   38h30m!     -45m      15:56 # <-- same here
```

> 15:12

```sh
             Total    Should     Diff   End-Time
Today        6h35m    6h25m!     +10m      15:02 # <-- ticked yet another minute
Other       31h10m    32h5m!     -55m
          ======== ========= ========
All         37h45m   38h30m!     -45m      15:57 # <-- dito
```

**Comment by jotaen** [comment]
This is expected, but I agree it might not be intuitive, so the UI could potentially be improved.

The reasoning is this:

- When using `--now`, klog behaves as if the `?` placeholder was substituted with the wall clock time “right now”. If you use `--follow` additionally, then it keeps doing that as time goes on. That way, you continuously “consume” time, and move closer to your should-total goal.
- If there is no open time range, then you don’t “consume” time, so the designated end-time moves forward with every minute that elapses – which makes sense, because you are not doing anything after all. E.g., if you have a time debt of `3h`, then it keeps pushing forward the end-time, until you start an activity.

UI-wise, the output could make it obvious whether there is actually an ongoing activity (open time range) that “consumes” time. I’ll think about it, but if you have ideas how the semantics could be improved to be more intuitive, please let me know.

**Comment by jotaen** [comment]
Ideas: Maybe there shouldn’t be any end-time at all, if there is no ongoing activity? Or it’s displayed in parentheses, or with subdued color?

**Comment by ulken** [comment]
As always, you manage to clear things up, so big props to you 👍

But you touch on a valid point, ideally it shouldn't need such in-depth explanations, but rather be self-explanatory.

Still I find it weird to have an end-time forecast when the should-total have already been reached (or even surpassed). 

I think it shouldn't show at all, or be capped.
